test/uk/gov/hmrc/customs/emailfrontend/config/connectors/httpparsers/EmailVerificationStateHttpParserSpec.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customs.emailfrontend.connectors.httpparsers
+
+import play.api.http.Status
+import uk.gov.hmrc.http.HttpResponse
+import uk.gov.hmrc.customs.emailfrontend.utils.SpecBase
+
+class EmailVerificationStateHttpParserSpec extends SpecBase {
+
+  "GetEmailVerificationStateHttpReads" should {
+
+    val httpParser = EmailVerificationStateHttpParser.GetEmailVerificationStateHttpReads
+
+    "successfully parse an OK response" in {
+      val httpResponse = HttpResponse(Status.OK, "")
+      val result = httpParser.read("GET", "/some/url", httpResponse)
+
+      result shouldBe Right(EmailVerificationStateHttpParser.EmailVerified)
+    }
+
+    "successfully parse a NOT_FOUND response" in {
+      val httpResponse = HttpResponse(Status.NOT_FOUND, "")
+      val result = httpParser.read("GET", "/some/url", httpResponse)
+
+      result shouldBe Right(EmailVerificationStateHttpParser.EmailNotVerified)
+    }
+
+    "successfully parse a BAD_REQUEST response" in {
+      val httpResponse = HttpResponse(Status.BAD_REQUEST, "Invalid request")
+      val result = httpParser.read("GET", "/some/url", httpResponse)
+
+      result shouldBe Left(EmailVerificationStateHttpParser.EmailVerificationStateErrorResponse(Status.BAD_REQUEST, "Invalid request"))
```

Comment:
Could you please split this line over multiple line as it is very long line. Better to split after Left(